File: admin-backend/src/api/v1/auth_api.py
"""
관리자/상담사 인증 API (로그인/로그아웃)
"""
from fastapi import APIRouter, Depends, Response, Request

from src.common.response import ok, APIResponse
from src.schemas.auth_schema import LoginRequest, LoginResponse, TokenPayload
from src.services.deps import (
    get_admin_service,
    get_counselor_service,
    get_auth_service,
)
from src.services.admin_service import AdminService
from src.services.counselor_service import CounselorService
from src.services.auth_service import AuthService
from src.common.utils.auth_utils import verify_admin_role, verify_counselor_role


router = APIRouter(prefix="/auth", tags=["auth"])


def _set_auth_cookies(response: Response, access_token: str, refresh_token: str) -> None:
    # Access Token (쿠키: 30분)
    response.set_cookie(
        key="access_token",
        value=access_token,
        httponly=True,
        secure=True,
        samesite="lax",
        max_age=30 * 60,
    )
    # Refresh Token (쿠키: 7일)
    response.set_cookie(
        key="refresh_token",
        value=refresh_token,
        httponly=True,
        secure=True,
        samesite="lax",
        max_age=7 * 24 * 60 * 60,
    )


def _clear_auth_cookies(response: Response) -> None:
    response.delete_cookie("access_token")
    response.delete_cookie("refresh_token")


@router.post("/admin/login", response_model=APIResponse[LoginResponse], summary="관리자 로그인")
async def admin_login(
    login_request: LoginRequest,
    response: Response,
    admin_service: AdminService = Depends(get_admin_service),
    auth_service: AuthService = Depends(get_auth_service),
):
    access_token, summary = await admin_service.login(login_request.user_id, login_request.password)
    refresh_token = auth_service.create_refresh_token(user_id=summary["user_id"], email=summary["email"], role="admin")

    _set_auth_cookies(response, access_token, refresh_token)

    return ok(
        data=LoginResponse(
            user_id=summary["user_id"],
            email=summary["email"],
            nickname=summary["nickname"],
            access_token_expires_in=30 * 60,
            refresh_token_expires_in=7 * 24 * 60 * 60,
        ),
        message="관리자 로그인 성공",
    )


@router.post("/admin/logout", summary="관리자 로그아웃")
async def admin_logout(response: Response):
    _clear_auth_cookies(response)
    return ok(data=None, message="관리자 로그아웃 성공")


@router.post("/counselor/login", response_model=APIResponse[LoginResponse], summary="상담사 로그인(관리자)")
async def counselor_login(
    login_request: LoginRequest,
    response: Response,
    counselor_service: CounselorService = Depends(get_counselor_service),
    auth_service: AuthService = Depends(get_auth_service),
):
    access_token, summary = await counselor_service.login(login_request.user_id, login_request.password)
    refresh_token = auth_service.create_refresh_token(user_id=summary["user_id"], email=summary["email"], role="counselor")

    _set_auth_cookies(response, access_token, refresh_token)

    return ok(
        data=LoginResponse(
            user_id=summary["user_id"],
            email=summary["email"],
            nickname=summary["nickname"],
            access_token_expires_in=30 * 60,
            refresh_token_expires_in=7 * 24 * 60 * 60,
        ),
        message="상담사 로그인 성공",
    )


@router.post("/counselor/logout", summary="상담사 로그아웃(관리자)")
async def counselor_logout(response: Response):
    _clear_auth_cookies(response)
    return ok(data=None, message="상담사 로그아웃 성공")

File: admin-front/src/api/auth/authApi.ts
import http from '@/api/_config/http';
import type { LoginRequest, LoginResponse, APIResponse } from '@/types/auth';

// admin-backend의 실제 라우팅 경로: /api/v1/auth
const AUTH_PREFIX = '/api/v1/auth';

/**
 * 관리자 로그인
 * - POST /auth/admin/login
 * - HttpOnly 쿠키로 access_token, refresh_token 자동 설정
 * @param loginData - 로그인 요청 데이터 (user_id, password)
 * @returns LoginResponse (user_id, email, nickname, token 만료 시간)
 */
export async function adminLogin(loginData: LoginRequest): Promise<APIResponse<LoginResponse>> {
  return http.post(`${AUTH_PREFIX}/admin/login`, loginData);
}

/**
 * 관리자 로그아웃
 * - POST /auth/admin/logout
 * - HttpOnly 쿠키의 토큰 삭제
 */
export async function adminLogout(): Promise<APIResponse<void>> {
  return http.post(`${AUTH_PREFIX}/admin/logout`);
}

/**
 * 상담사 로그인
 * - POST /auth/counselor/login
 * - HttpOnly 쿠키로 access_token, refresh_token 자동 설정
 */
export async function counselorLogin(loginData: LoginRequest): Promise<APIResponse<LoginResponse>> {
  return http.post(`${AUTH_PREFIX}/counselor/login`, loginData);
}

/**
 * 상담사 로그아웃
 * - POST /auth/counselor/logout
 * - HttpOnly 쿠키의 토큰 삭제
 */
export async function counselorLogout(): Promise<APIResponse<void>> {
  return http.post(`${AUTH_PREFIX}/counselor/logout`);
}
